### PR TITLE
Vandit move room name

### PR DESCRIPTION
**Pull request by VanditPatel1**
Here an example of using roomName and setRoomName in context form, I think I changed all occurrences of it but in case I missed one, plz change it. 

I tried flow of making a room with 2 players and trying some moves, it didn't break so I'm assuming its good.

### CODE REVIEW

**Review by Shehan29 — APPROVED** [review]

File: src/gameplay/GameStart.js [hunk 1]
```diff
@@ -89,7 +93,7 @@ function GameStart(props) {
     } else {
       console.log(props.playerID);
       return (
-        <WaitForHost roomName={props.roomName} id = {props.playerID} playerArray = {playerIDs} playerNames = {players} setPlayerIndex = {props.setPlayerIndex} setPlayerNames = {props.setPlayerNames}/>
+        <WaitForHost id = {props.playerID} playerArray = {playerIDs} playerNames = {players} setPlayerIndex = {props.setPlayerIndex} setPlayerNames = {props.setPlayerNames}/>
```
Comment (preetshah123) [inline src/gameplay/GameStart.js:96]: why arent you passing in roomname here but you are later in the JoinGame tag

File: src/gameplay/GameStart.js [hunk 2]
```diff
@@ -101,8 +105,8 @@ function GameStart(props) {
 	        <li className="list-group-item" key={name}>{name} is in the lobby</li>
 	      ))}
 	    </ol>
-      <div align="center"> <h4>Current Room: {props.roomName}</h4> </div>
-	    <JoinGame isHost={props.isHost} roomName={props.roomName} playerID = {props.playerID} setPlayerIndex = {props.setPlayerIndex} setPlayerNames = {props.setPlayerNames}/>
+      <div align="center"> <h4>Current Room: {roomName}</h4> </div>
+	    <JoinGame isHost={props.isHost} roomName={roomName} playerID = {props.playerID} setPlayerIndex = {props.setPlayerIndex} setPlayerNames = {props.setPlayerNames}/>
```
Comment (preetshah123) [inline src/gameplay/GameStart.js:109]: here

Comment (VanditPatel1) [reply src/gameplay/GameStart.js:109]: oh ya here I can not pass it in and use context too, I just missed this one

Comment (preetshah123) [reply src/gameplay/GameStart.js:109]: thought so too, cool

Comment (VanditPatel1) [reply src/gameplay/GameStart.js:109]: either way works, actually I think I left it cause this component is held within GameStart component so passing it as a prop made sense, the other component is in another file so I used context there. TBH either works tho

File: src/App.js [hunk 3]
```diff
@@ -47,27 +43,30 @@ function App() {
...
-              <MainGameScreen playerID={playerID} isHost={isHost} roomName={roomName} playerIndex={playerIndex} playerNames = {playerNames}/>
-            </div>
-          </Route>
-          <Route exact path="/GameStart">
-            <div align="center">
-              <GameStart playerID={playerID} isHost={isHost} roomName={roomName} setPlayerIndex={setPlayerIndex} setPlayerNames = {setPlayerNames}/>
-            </div>
-          </Route>
-        </Switch>
+        <RoomContextProvider>
```
Comment (preetshah123) [inline src/App.js:51]: So if we have multiple contexts, how would it look? Like would it be nested conext tags? 
Like: 
<playerIdContext>
      <RoomContext> etc

Comment (VanditPatel1) [reply src/App.js:51]: ya you could nest

Comment (VanditPatel1) [reply src/App.js:51]: player context is also only needed in a few places so we could just wrap those

**Pull request merged**